clcache.py

```diff
@@ -1246,15 +1267,23 @@ def printStatistics(cache):
             stats.numCallsWithPch(),
         ))
 
+
 def resetStatistics(cache):
-    with cache.statistics as stats:
+    with cache.statistics.lock, cache.statistics as stats:
         stats.resetCounters()
     print('Statistics reset')
 
+
 def cleanCache(cache):
     with cache.statistics as stats, cache.configuration as cfg:
-        cache.clean(stats, cfg.maximumCacheSize())
-    print('Cache cleaned')
+        currentCacheSize = stats.currentCacheSize()
+        maximumCacheSize = cfg.maximumCacheSize()
+
+    if currentCacheSize >= maximumCacheSize:
+        with cache.lock, cache.statistics as stats:
+            cache.clean(stats, maximumCacheSize)
+        print('Cache cleaned')
```

Comment:
I would continue printing the success message in case cache was already small enough. Changing this behavior (e.g. by adding different messages) is out of scope I think.